### DISCUSSION TITLE
Are there plans for increased functionality of Malt's texture nodes?

### DISCUSSION THREAD

**Discussion by DaylightIsJustABotanist** [bnpr/Malt#430] [Feedback & Feature Requests]
Nodes like voronoi, wave, and noise textures in EEVEE have a number of options related to the number of dimensions as well as, among other options such as voronoi randomness that Malt currently does not. Are these additional features planned to be implemented at some point?

**Comment by Kolupsy** [comment]
The wave texture - in terms of supported features - should match the one presented in Eevee and cycles (at least in more recent versions of malt). What would want from the wave texture that it currently does not have? The noise and voronoi already have variations for lower dimensions but the implementation for supporting nodes with more than one dropdown is not entirely clear yet, that's why they are hidden for now. You can still get access to them through the "Internal" node category. Randomness for voronoi and different voronoi exponents are not yet supported.

**Reply by DaylightIsJustABotanist** [reply]
For the wave texture,  in EEVEE I am seeing functions for distortion as well as bands or rings that I am not seeing in Malt. EEVEE also has the ability to easily change the direction waves are facing but that can be done with mapping in Malt so it's not as important. I would also like to see the ability to adjust wave spacing in Malt, distinct from scale in the sense that it could be randomized without also distorting the waves themselves, but as I write this I realize that it can probably be done with mapping nodes.

For the additional dimension support, could this be enabled with WXYZ checkboxes or int 0-3 rather than as a second dropdown? The current, on voronoi for instance, dropdown with infinite or tiled does not have an input either so using either a set of boolean inputs or an integer one could increase functionality by allowing multiple nodes to change dimensions at once, or be changed from outside a node group.
